src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1288,6 +1293,83 @@ void SolverInterfaceImpl::readScalarData(
   PRECICE_DEBUG("Read value = " << value);
 }
 
+
+int SolverInterfaceImpl::setBoundingBox(
+    const int     meshID,
+    const double *boundingBox) const
+{
+  PRECICE_TRACE(meshID);
+  PRECICE_ASSERT(boundingBox != nullptr);
+  PRECICE_REQUIRE_MESH_MODIFY(meshID);
+
+  // Get the related mesh
+  MeshContext & context = _accessor->meshContext(meshID);
+  mesh::PtrMesh mesh(context.mesh);
+  PRECICE_DEBUG("Define bounding box");
+  // Transform bounds into a suitable format
+  int                 dim = mesh->getDimensions();
+  std::vector<double> bounds(dim * 2);
+
+  for (int d = 0; d < dim; ++d) {
+    // Assert that min is lower or equal to max
+    PRECICE_ASSERT(boundingBox[2 * d] <= boundingBox[2 * d + 1]);
+    bounds[2 * d]     = boundingBox[2 * d];
+    bounds[2 * d + 1] = boundingBox[2 * d + 1];
+  }
+  // Create a bounding box
+  mesh::BoundingBox providedBoundingBox(bounds);
+  // Expand the mesh associated bounding box
+  mesh::BoundingBox &bbox = mesh->getBoundingBox();
+  bbox.expandBy(providedBoundingBox);
```

Comment:
We are actually setting the BB. Would be easier to have a `mesh::setBoundingBox`, right?

Reply:
To be decided. I guess both could be useful.